warp/src/raygun/mod.rs

```diff
@@ -548,41 +548,54 @@ impl Default for ConversationSettings {
 #[repr(C)]
 pub struct DirectConversationSettings {}
 
-#[derive(Default, Debug, Copy, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Display)]
-#[display(
-    fmt = "Everyone can add participants: {}, Everyone can change name: {}",
-    "if self.members_can_add_participants {\"✅\"} else {\"❌\"}",
-    "if self.members_can_change_name {\"✅\"} else {\"❌\"}"
-)]
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Display)]
+#[display(fmt = "Roles: {:?}", "self.roles")]
 #[repr(C)]
 pub struct GroupSettings {
-    // Everyone can add participants, if set to `true``.
-    #[serde(default)]
-    members_can_add_participants: bool,
-    // Everyone can change the name of the group.
+    // Roles to manage permissions. Key is the unique name of the role.
     #[serde(default)]
-    members_can_change_name: bool,
+    roles: IndexMap<String, Role>,
+}
+
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Debug)]
+pub struct Role {
+    // Participants assigned to this role
+    participants: Vec<DID>,
+    // Permissions of this role
+    permissions: Vec<Permission>,
+}
```

Comment:
I wonder if we should reduce this to `IndexMap<DID, IndexSet<Permission>>` (or some variant of map/set)? Thoughts on that?

Reply:
done 👍